src/uu/mknod/src/mknod.rs

```diff
@@ -66,18 +70,15 @@ pub struct Config<'a> {
 fn mknod(file_name: &str, config: Config) -> i32 {
     let c_str = CString::new(file_name).expect("Failed to convert to CString");
 
-    // the user supplied a mode
-    let set_umask = config.mode & MODE_RW_UGO != MODE_RW_UGO;
-
     unsafe {
-        // store prev umask
-        let last_umask = if set_umask { libc::umask(0) } else { 0 };
+        // set umask to 0 and store previous umask
+        let prev_umask = if config.use_umask { 0 } else { libc::umask(0) };
```

Comment:
I think we can improve readability by storing the umask as an Option, say:

let prev_umask_option = if config.use_umask { None } else { Some(libc::umask(0)) };

 then we can just do

`if let Some(prev_umask) = prev_umask_option`

or something below which makes the control flow easier to follow as the decision on the config is in one place and the logic "did we change it? we need to restore it" is more cleanly communicated.